core/src/main/java/org/infinispan/remoting/transport/impl/JGroupsRequestTracing.java

```diff
@@ -0,0 +1,100 @@
+package org.infinispan.remoting.transport.impl;
+
+import static org.infinispan.util.logging.Log.CONTAINER;
+
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.commands.ReplicableCommand;
+import org.infinispan.commands.remote.SingleRpcCommand;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.infinispan.factories.AbstractComponentFactory;
+import org.infinispan.factories.AutoInstantiableFactory;
+import org.infinispan.factories.annotations.DefaultFactoryFor;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.factories.annotations.Start;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.registry.InternalCacheRegistry;
+import org.infinispan.security.actions.SecurityActions;
+import org.infinispan.telemetry.InfinispanSpan;
+import org.infinispan.telemetry.InfinispanSpanAttributes;
+import org.infinispan.telemetry.InfinispanTelemetry;
+import org.infinispan.telemetry.SpanCategory;
+import org.infinispan.telemetry.impl.DisabledInfinispanSpan;
+import org.infinispan.xsite.SingleXSiteRpcCommand;
+
+@Scope(Scopes.GLOBAL)
+public class JGroupsRequestTracing {
+
+   @Inject
+   protected InfinispanTelemetry telemetryService;
+
+   @Inject
+   protected EmbeddedCacheManager cacheManager;
+
+   @Inject
+   protected InternalCacheRegistry internalCacheRegistry;
+
+   @Inject
+   protected GlobalConfiguration configuration;
+
+   private boolean globalTracingEnabled;
```

Comment:
This usage won't work here as this is not runtime enablable. You need to either register a change listener with the Configuration that I have mentioned quite a few times already or completely remove this variable as this class will never see the update to the tracing configuration.

Reply:
this variable it is not supposed to be runtime enablable